src/react-app-env.d.ts

```diff
@@ -7,7 +7,10 @@ declare module '@metamask/jazzicon' {
 declare module 'fortmatic'
 
 interface Window {
+  WalletLinkProvider?: any
```

Comment:
sorry can you just add some comments? when is WalletLinkProvider on the window, when is walletLinkExtension on the window, when is window.ethereum on the window with isCoinbaseWallet=true

Reply:
the code below is very hard to understand, especially not knowing why there are 3 different ways for coinbase to be injected to the window

Reply:
that's fair. added docs.